keps/sig-node/127-user-namespaces/README.md

```diff
@@ -1046,8 +1118,144 @@ details). For now, we leave it here.
+
+        We can add a metric for this on general, but as any pod creation that fails has this direct
+        feedback to the user, we don't think it is needed.
+
+  - Mitigations: What can be done to stop the bleeding, especially for already
+    running user workloads?
+
+        Remove the pod.spec.HostUsers field or disable the feature gate.
+
+  - Diagnostics: What are the useful log messages and their required logging
+    levels that could help debug the issue?
+    Not required until feature graduated to beta.
+
+        No extra logs, the error is returned to the user.
+
+  - Testing: Are there any tests for failure mode? If not, describe why.
+
+        Yes, there are unit tests for this at the pkg/kubelet/userns package.
+
+- Error to save/read pod mappings file.
+The kubelet records the mappings used for a pod in a file. There can be an error while reading or
+writing to this file.
+
+  - Detection: How can it be detected via metrics? Stated another way:
+    how can an operator troubleshoot without logging into a master or worker node?
+
+        Errors are returned to the operation failed (like pod creation), no need to see metrics nor
+        logs.
+
+        Errors are returned to the either on:
+         * Kubelet initialization: the initialization fails if the feature gate is active and there is a
+           fatal error while reading the pod's mapping file.
+         * Pod creation: an error is returned to the user (shown in the pod events) if there is an error to
+           read or write to this file.
+
+        We can add a metric for this, but as the errors have very direct feedback (kubelet init
+        fails with a clear error or a pod creation fails with a clear error), we don't think it is needed.
+
+  - Mitigations: What can be done to stop the bleeding, especially for already
+    running user workloads?
+
+        Remove the pod.spec.HostUsers field or disable the feature gate.
+
+  - Diagnostics: What are the useful log messages and their required logging
+    levels that could help debug the issue?
+    Not required until feature graduated to beta.
+
+        The most likely reason why this can fail is if the kernel doesn't support idmapped mounts.  In this
+        case the error reported from the OCI runtime is clear enough, both the runc and crun OCI runtimes were
+        instrumented to return a clear error message.
+
+  - Testing: Are there any tests for failure mode? If not, describe why.
+
+        There are extensive unit tests on the function that parses the file content.
+        There are no tests for failures to read or write the file, the code-paths just return the errors
+        in those cases.
+
+
+- Error getting the kubelet IDs range configuration
+  - Detection: How can it be detected via metrics? Stated another way:
+    how can an operator troubleshoot without logging into a master or worker node?
+
+        In this case the Kubelet will fail to start with a clear error message.
```

Comment:
Assuming this is through subuids binary don't we fallback to default range?

Reply:
I think we should fallback to default ranges only if there is no `kubelet` user configured, or there is no `getsubids` binary present.

If the `kubelet` user is defined and we fail to read its configuration, an error should be reported.